build_scripts/android/build.sh

```diff
@@ -25,14 +36,28 @@ cd -
 # If NDK_ROOT is not set or is the wrong version, use to the version in /tmp.
 if [[ -z "${NDK_ROOT}" || ! $(grep -q "Pkg\.Revision = 16\." "${NDK_ROOT}/source.properties") ]]; then
     if [[ ! -d /tmp/android-ndk-r16b ]]; then
-	echo "NDK r16b not present in /tmp, please run install_prereqs.sh script."
+	echo "Recommended NDK version r16b not present in /tmp."
+	if [[ ! -z "${stl}" ]]; then
+	    echo "STL may only be specified if using the recommended NDK version."
+	    echo "Please run install_prereqs.sh script and try again."
+	    exit 2
+	else
+	    echo "Please run install_prereqs.sh if you wish to use the recommended NDK version."
+	    echo "Continuing with default NDK..."
+	    sleep 2
+	fi
     fi
     export NDK_ROOT=/tmp/android-ndk-r16b
+    export ANDROID_NDK_HOME=/tmp/android-ndk-r16b
 fi
-set -ex
 cd "${sourcepath}"
-./gradlew assembleRelease
-set +x
+set +e
+for retry in {1..10} error; do
```

Comment:
Could you comment why we are doing the retries?

Reply:
Done.